### PR TITLE
NativeFunctionInvocationFixer - add option to remove redundant backslashes

### PR DESCRIPTION
**Pull request by kubawerlos**
If we add it to `@Symfony` rule set with `true` then in Symfony v4.1.6 26 files would be affected, and here - at master - 6 files.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -916,6 +916,8 @@ Choose from the list of available rules:
     defaults to ``['@internal']``
   - ``scope`` (``'all'``, ``'namespaced'``): only fix function calls that are made
     within a namespace or fix all; defaults to ``'all'``
+  - ``strict`` (``bool``): whether leading ``\`` from function class not meant to have
```
Comment (SpacePossum) [inline README.rst]: >from function class

I think this should be `from function call` ?

**Comment by keradus** [comment]
> If we add it to @Symfony rule set with true then in Symfony v4.1.6 26 files would be affected, and here - at master - 6 files.

@nicolas-grekas , WDYT ?

**Comment by nicolas-grekas** [comment]
We could enable it, so that things have only one valid way to be written.

**Comment by keradus** [comment]
:+1:

@kubawerlos , can you update ruleset definition, in that case ?

**Comment by kubawerlos** [comment]
@keradus updated

@nicolas-grekas should I make PR to Symfony to remove extra `\` to avoid surprise when this is released? If so what branch to target?

**Comment by nicolas-grekas** [comment]
no need, let's wait for the fabbot update and we'll do it in a batch starting from 2.8

File: src/Fixer/ConstantNotation/NativeConstantInvocationFixer.php [hunk 2]
```diff
@@ -115,44 +115,44 @@ public function configure(array $configuration = null)
...
                 unset($constantsToEscape[$constantIndex]);
             }
         }
 
-        $caseInsensitiveConstantsToEscape = \array_diff(
-            \array_unique($caseInsensitiveConstantsToEscape),
-            \array_map('strtolower', $uniqueConfiguredExclude)
+        $caseInsensitiveConstantsToEscape = array_diff(
+            array_unique($caseInsensitiveConstantsToEscape),
+            array_map('strtolower', $uniqueConfiguredExclude)
```
Comment (keradus) [inline src/Fixer/ConstantNotation/NativeConstantInvocationFixer.php]: :(

**Comment by SpacePossum** [comment]
@kubawerlos friendly heads up about the conflict here

**Comment by kubawerlos** [comment]
@SpacePossum update, I had to remove it from Travis's "strict" job as it was conflicting with the `@Symfony` configuration. Why did we had it there anyway?

**Comment by SpacePossum** [comment]
The test was added so that we will know that if we would make all function calls within the project native calls (were applicable) we would still be good, as we currently do not enforce this.

I think this is still a good idea though, but I see the trouble arise here, therefore I need some time to think about removing it as proposed in the PR.

**Comment by kubawerlos** [comment]
@SpacePossum we can create separate step to apply these rules and run only tests or (more hacky) update the config with `native_function_invocation` to override the one from `@Symfony`

**Comment by SpacePossum** [comment]
@kubawerlos  please add tests @ `NativeConstantInvocationFixerTest` to cover the new configuration option, than I think this PR is good to go :)

About the test, if I run the original fix command and than the unit tests the test passes. So I'm not sure where this would fail now?

**Comment by SpacePossum** [comment]
Sorry for being slow; please ignore my first remark and about the test; I agree with your previous suggestion; 

>we can create separate step to apply these rules and run only tests

**Comment by keradus** [comment]
I had drop `native_function_invocation` from migration again.
We had it to see if we run on optimized build.
Now, definition of `what` optimized means changed due to better understanding of `\` resolution, and default setting of it is not what we now consider as best practice. `native_function_invocation` we have in our config we use internally is enough for CI.

**Comment by keradus** [comment]
Thank you @kubawerlos.

**Pull request merged**